### PR TITLE
fs: cleanup uv__fs_scandir

### PR DESCRIPTION
**Pull request by santigimeno**
The `out` section is only reachable now if `scandir` returns 0.

### CODE REVIEW

File: src/unix/fs.c [hunk 1]
```diff
@@ -389,30 +388,18 @@ static ssize_t uv__fs_scandir(uv_fs_t* req) {
...
-  if (n == 0)
-    goto out; /* osx still needs to deallocate some memory */
-  else if (n == -1)
+  if (n == 0) {
+    /* osx still needs to deallocate some memory */
+    if (dents) {
+      free(dents);
+      dents = NULL;
+    }
+  } else if (n == -1)
```
Comment (cjihrig) [inline src/unix/fs.c]: The `else if` should have curly braces now I think.

Comment (saghul) [inline src/unix/fs.c]: It doesn't *need* them, but I'd say it's good to add them.

**Comment by santigimeno** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/195/

File: src/unix/fs.c [hunk 2]
```diff
@@ -389,28 +388,17 @@ static ssize_t uv__fs_scandir(uv_fs_t* req) {
   /* NOTE: We will use nbufs as an index field */
   req->nbufs = 0;
 
-  if (n == 0)
-    goto out; /* osx still needs to deallocate some memory */
-  else if (n == -1)
+  if (n == 0) {
+    /* osx still needs to deallocate some memory */
+    if (dents) {
+      free(dents);
```
Comment (bnoordhuis) [inline src/unix/fs.c]: Can you add the `/* Memory was allocated using the system allocator, so use free() here. */` comment here?

**Comment by cjihrig** [comment]
CI failures seem to be all of the usual suspects.

File: src/unix/fs.c [hunk 3]
```diff
@@ -389,28 +388,17 @@ static ssize_t uv__fs_scandir(uv_fs_t* req) {
...
+      dents = NULL;
+    }
```
Comment (bnoordhuis) [inline src/unix/fs.c]: You don't strictly need to wrap the call to `free(dents)` in a NULL pointer check, `free(NULL)` is a no-op.

**Comment by santigimeno** [comment]
Updated with @bnoordhuis comments

**Comment by saghul** [comment]
Landed in d251dee, thanks!

**Comment by saghul** [comment]
Oh, I just noticed it was you Santi, sorry about that!

**Comment by santigimeno** [comment]
np, thanks!

**Pull request closed without merging**